CUDADataFormats/Vertex/interface/ZVertexSoAHeterogeneousHost.h

```diff
@@ -0,0 +1,28 @@
+#ifndef CUDADataFormats_Vertex_ZVertexHeterogeneousHost_H
+#define CUDADataFormats_Vertex_ZVertexHeterogeneousHost_H
+
+#include <cstdint>
+
+#include "CUDADataFormats/Vertex/interface/ZVertexUtilities.h"
+#include "CUDADataFormats/Common/interface/PortableHostCollection.h"
+
+// TODO: The class is created via inheritance of the PortableHostCollection.
+// This is generally discouraged, and should be done via composition.
+// See: https://github.com/cms-sw/cmssw/pull/40465#discussion_r1067364306
+template <int32_t S>
+class ZVertexSoAHeterogeneousHost : public cms::cuda::PortableHostCollection<ZVertexSoAHeterogeneousLayout<>> {
+public:
+  explicit ZVertexSoAHeterogeneousHost() : cms::cuda::PortableHostCollection<ZVertexSoAHeterogeneousLayout<>>(S) {}
+
+  // Constructor which specifies the SoA size and CUDA stream
+  explicit ZVertexSoAHeterogeneousHost(cudaStream_t stream)
+      : PortableHostCollection<ZVertexSoAHeterogeneousLayout<>>(S, stream) {}
+};
+
+namespace zVertex {
```

Comment:
Do we really need to next the `ZVertexSoAHost` alias inside the `zVertex` namespace ?

Then we end up using it as `zVertex::ZVertexSoAHost` which seems redundant.

Reply:
@fwyzard I removed the `zVertex` namespace. Should this also be done with the `pixelTrack` namespace in `TrackSoAHeterogeneousDevice.h` and `TrackSoAHeterogeneousHost.h`? I think those are a bit less redundant, since it is not exactly the same name, but still there might not be any advantage on expliciting it that way...

Reply:
I'd keep the `pixelTrack` namespace, because it is used to group the free functions in `CUDADataFormats/Track/interface/PixelTrackUtilities.h`